refactor(mount): Refactor get self quota

rolysr opened the pull request:
Previous changes introduced in commit `a1db23de` caused the master server to malfunction with clients at or below the `v4.9.0` release. This commit refactors the code for the self quota retrieval functions so that no changes break the client–master communication protocol, while maintaining the previously introduced functionality and ensuring compatibility with older client versions.

uristdwarf commented on src/mount/mastercomm.cc:
```diff
@@ -3135,33 +3135,35 @@ uint8_t fs_get_self_quota(uint32_t uid, uint32_t gid, uint32_t inode, std::vecto
-			return SAUNAFS_STATUS_OK;
-		} else {
-			return SAUNAFS_ERROR_EINVAL;
+			matocl::fuseGetQuota::deserialize(message, rec->packetId, status);
+			throw Exception(": failed", status);
```
I would also considering the function name where this exception originated

uristdwarf commented:
I think the solution should be making the new protocol you added in the previous commit work with the old protocol, by adding a new version of that protocol

rolysr committed:
refactor(mount): Refactor get self quota

Previous changes introduced in commit `a1db23de` caused the master
server to malfunction with clients at or below the `v4.9.0` release.
This commit refactors the code for the self quota retrieval related
code so that no changes break the client–master communication
protocol, while maintaining the previously introduced functionality
and ensuring compatibility with older client versions.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

dmga44 reviewed (approved):
LGTM.
